Userland/Libraries/LibTLS/Certificate.h

```diff
@@ -18,36 +18,92 @@
 
 namespace TLS {
 
-enum class CertificateKeyAlgorithm {
+enum class CertificateKeyAlgorithm : u8 {
     Unsupported = 0x00,
     RSA_RSA = 0x01,
+    RSA_MD2 = 0x2,
+    RSA_MD4 = 0x3,
     RSA_MD5 = 0x04,
     RSA_SHA1 = 0x05,
+    RSA_OAEP = 0x6,
+    RSAES_OAEP = 0x7,
+    RSA_MGF1 = 0x8,
+    RSA_SPECIFIED = 0x9,
+    RSA_PSS = 0xa,
     RSA_SHA256 = 0x0b,
     RSA_SHA384 = 0x0c,
     RSA_SHA512 = 0x0d,
+    RSA_SHA224 = 0xe,
+    ECDSA_SHA224 = 0x10,
+    ECDSA_SHA256 = 0x11,
+    ECDSA_SHA384 = 0x12,
+    ECDSA_SHA512 = 0x13,
+    ECDSA_SECP256R1 = 0x14,
+    ECDSA_SECP384R1 = 0x15,
+};
+
+struct BasicConstraints {
+    bool is_certificate_authority;
+    Crypto::UnsignedBigInteger path_length_constraint;
+};
+
+class RelativeDistinguishedName {
+public:
+    ErrorOr<String> to_string();
+
+    ErrorOr<AK::HashSetResult> try_set(String key, String value)
+    {
+        return m_members.try_set(key, value);
+    }
+
+    Optional<String> get(String key)
```

Comment:
Can we get an overload here for `Optional<String> get(StringView key)`?

Another nice thing would be an Enum (?) overload to not have to search the OID if you want to access a field.

Reply:
Changed get() to take a StringView, and also added enums for some of the common types 👍